vspace/src/main/java/edu/asu/diging/vspace/web/staff/DownloadFileController.java

```diff
@@ -0,0 +1,45 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.io.IOException;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.diging.vspace.core.model.IVSFile;
+import edu.asu.diging.vspace.core.services.impl.FileManager;
+
+@Controller
+public class DownloadFileController {
+    
+    @Autowired
+    private FileManager fileManager;
+    
+    
+    @RequestMapping(value = "/staff/files/download/{fileId}", method = RequestMethod.GET )
+    public ResponseEntity<Resource> downloadFile(Model model, @PathVariable String fileId) throws IOException {
+        Resource resource = null;  
+        IVSFile file= null;
+        try {        
+            file = fileManager.getFileById(fileId);
+            resource = fileManager.downloadFile(file.getFilename());                  
+        } catch (IOException e) {
+            return new ResponseEntity<Resource>(resource, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        return ResponseEntity.ok()
+                .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + file.getFilename())
+                .contentLength(resource.contentLength())
+                .contentType(MediaType.APPLICATION_OCTET_STREAM)
```

Comment:
this will always make the browser download the file. the content-header should be the correct of the file.

Reply:
@jdamerow, Do you mean the content type should be the actual file type like pdf / text etc?

Reply:
it should be the content type of the file (e.g. text/plain, application/pdf, etc).